networking_nsxv3/api/rpc.py

```diff
@@ -231,69 +170,82 @@ def plugin(self):
         return self._plugin
 
     @log_helpers.log_method_call
-    def get_port_revision_tuples(self, context, host, limit, created_after):
-        return db.get_port_revision_tuples(context, host, limit, created_after)
+    def get_ports_revisions(self, context, host, limit, offset):
+        return db.get_ports_revisions(context, host, limit, offset)
 
     @log_helpers.log_method_call
-    def get_qos_policy_revision_tuples(self, context, limit, created_after):
-        return db.get_qos_policy_revision_tuples(context, limit, created_after)
-
+    def get_qoses_revisions(self, context, host, limit, offset):
+        return db.get_qoses_revisions(context, host, limit, offset)
+    
     @log_helpers.log_method_call
-    @oslo_messaging.expected_exceptions(exceptions.ObjectNotFound)
-    def get_security_group_revision(self, context, security_group_id):
-        return db.get_security_group_revision(context, security_group_id)
+    def get_security_groups_revisions(self, context, host, limit, offset):
+        return db.get_security_groups_revisions(context, host, limit, offset)
 
     @log_helpers.log_method_call
-    def get_security_group_revision_tuples(
-            self, context, limit, created_after):
-        return db.get_security_group_revision_tuples(
-            context, limit, created_after)
-
-    def has_security_group_tag(
-            self, context, security_group_id, tag_name):
-        return db.has_security_group_tag(
-            context, security_group_id, tag_name)
+    def get_security_group(self, context, security_group_id):
+        id_rev = db.get_security_group_revision(context, security_group_id)
+        if id_rev:
+            return {
+                "id": id_rev[0],
+                "revision_number": id_rev[1],
+                "tags": db.get_security_group_tag(context, security_group_id),
+                "ports": db.get_port_id_by_sec_group_id(context, sec_group_id)
+            }
 
     @log_helpers.log_method_call
-    @oslo_messaging.expected_exceptions(exceptions.ObjectNotFound)
-    def get_qos(self, context, qos_id):
-        return db.get_qos(context, qos_id)
+    def get_rules_for_security_groups_id(self, context, security_group_id):
+        return db.get_rules_for_security_groups_id(context, security_group_id)
 
     @log_helpers.log_method_call
-    def get_qos_bwl_rules(self, context, qos_id):
-        return db.get_qos_bwl_rules(context, qos_id)
+    def get_security_group_members_ips(self, context, security_group_id):
+        return \
+            db.get_security_group_members_ips(context, security_group_id) + \
+            db.get_security_group_members_address_bindings_ips(context, security_group_id)
```

Comment:
I am a bit conflicted about that call, as  it is named exactly like the first db api call, but also does a second api call.
I would suggest renaming it.

I am suprised though that you have to do two independent db api calls to get all the IPs you want. I have to look deeper into the calls to understand it to make a more qualified comment. Personally, I think there should be a single one, but that may be a topic for a later change.

Reply:
- [ ] Research how to use single query for SG members all IPs

Reply:
@fwiesel merging both queries into a single database query will be simply concatenation of two results. I do not see any benefit to make this change. Let me know if you see this differently.

Reply:
Combining queries into one is indeed more a minor improvement. It saves a single round-trip, so fractions of milliseconds. And the concatenation will be done in the DB instead in python, which should be a bit faster. I would not say, that needs to be done, just a suggestion.

My main trouble is the naming. It is named the same as the first db api call, but it does something different, as does a second api call to the db. I would prefer a different name reflecting the differences.

Reply:
200: will be renamed to `get_security_group_members_effective_ips`